packages/db/src/upsert-resources.ts

```
@@ -151,5 +176,50 @@ export const upsertResources = async (
+
+    if (!_.isEqual(variables, rVariables)) return true;
+
+    const existingMetadata = _(existing.metadata)
+      .map((v) => [v.key, v.value])
+      .fromPairs()
+      .value();
+
+    if (!_.isEqual(existingMetadata, r.metadata)) return true;
+
+    const pickedAndOmitted = _.pick(existing, Object.keys(r));
+    delete pickedAndOmitted.metadata;
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)